src/bot/commands/approve.ts

```diff
@@ -274,86 +411,47 @@ export default class ApproveCommand extends Command {
-      const suggestionLink = messageLink(interaction.guild.id, suggestionsChannelId, proposal.message_id!);
-      const correctionLink = messageLink(
-        interaction.guild.id,
-        correctionsChannelId,
-        proposal.corrections[0].message_id!
-      );
-      return interaction.reply(
-        interactionInfo(`
-          Le nombre d'approbations requises pour l'ajout de [cette suggestion](${suggestionLink}) a déjà été atteint, seul [cette correction](${correctionLink}) nécessite encore des approbations.`)
-      );
-    }
-
-    const godfathers = await renderGodfatherLine(interaction, proposal);
-
-    const field = embed.fields?.[embed.fields.length - 1];
-    if (field) {
-      const { base, correction } = field.value.match(dataSplitRegex)!.groups!;
-      field.value = [base, correction, godfathers].filter(Boolean).join('\n\n');
-    } else {
-      const { base, correction } = embed.description!.match(dataSplitRegex)!.groups!;
-      embed.description = [base, correction, godfathers].filter(Boolean).join('\n\n');
-    }
+    const newembed = await updateProposalsEmbed(interaction, proposal, embed);
 
     await interaction.client.votes.deleteUserVotes(message, interaction.user.id);
 
-    if (proposal.approvals.length < neededApprovalsCount) {
-      await message.edit({ embeds: [embed] });
+    if (proposal.approvals.length < neededReportsApprovals) {
+      await message.edit({ embeds: [newembed] });
 
-      return interaction.reply(interactionValidate(`Votre [approbation](${message.url}) a été prise en compte !`));
+      await interaction.reply(interactionValidate(`Votre [approbation](${message.url}) a été prise en compte !`));
+      return null;
     }
 
     await interaction.deferReply({ ephemeral: true });
-
-    try {
-      switch (proposal.type) {
-        case ProposalType.SUGGESTION: {
-          await this.approveSuggestion(interaction, proposal, message, embed);
-          break;
-        }
-        case ProposalType.REPORT: {
-          await this.approveReport(interaction, proposal, message, embed);
-          break;
-        }
-        case ProposalType.CORRECTION: {
-          await this.approveCorrection(interaction, proposal, message, embed);
-          break;
-        }
-      }
-    } catch (error) {
-      console.error(error);
-      await interaction.editReply(
-        interactionProblem(
-          `Une erreur s'est produite lors de l'approbation de la [suggestion](${message.url}), veuillez contacter le développeur !`
-        )
-      );
-    }
+    return {
+      proposal: proposal,
+      embed: newembed
+    };
   }
 
-  async approveSuggestion(
-    interaction: MessageContextMenuCommandInteraction,
-    proposal: Suggestion,
-    message: Message,
-    embed: APIEmbed,
-    automerge = false
-  ): Promise<void> {
+  async approveSuggestion(options: ApproveOptions<Suggestion> & { automerge?: boolean }): Promise<void> {
```

Comment:
Déjà inclu dans le typage de ApproveOptions
```suggestion
  async approveSuggestion(options: ApproveOptions<Suggestion>): Promise<void> {
```

Reply:
Options à inclure directement
```suggestion
  async approveSuggestion({ interaction, proposal, message, embed, automerge }: ApproveOptions<Suggestion>): Promise<void> {
```